homeassistant/components/moehlenhoff_alpha2/climate.py

```diff
@@ -0,0 +1,169 @@
+    def hvac_action(self) -> str:
+        """Return the current running hvac operation."""
+        if not self.coordinator.data[self.heat_area_id]["_HEATCTRL_STATE"]:
+            return CURRENT_HVAC_IDLE
+        if self.coordinator.get_cooling():
+            return CURRENT_HVAC_COOL
+        return CURRENT_HVAC_HEAT
+
+    @property
+    def target_temperature(self) -> float:
+        """Return the temperature we try to reach."""
+        return float(self.coordinator.data[self.heat_area_id].get("T_TARGET", 0.0))
+
+    async def async_set_temperature(self, **kwargs) -> None:
+        """Set new target temperatures."""
+        target_temperature = kwargs.get(ATTR_TEMPERATURE)
+        if target_temperature is None:
+            return
+
+        await self.coordinator.async_set_target_temperature(
+            self.heat_area_id, target_temperature
+        )
+        self.coordinator.data[self.heat_area_id]["T_TARGET"] = target_temperature
+
+    @property
+    def preset_mode(self) -> str:
+        """Return the current preset mode."""
+        if self.coordinator.data[self.heat_area_id]["HEATAREA_MODE"] == 1:
+            return PRESET_DAY
+        if self.coordinator.data[self.heat_area_id]["HEATAREA_MODE"] == 2:
+            return PRESET_NIGHT
+        return PRESET_AUTO
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new operation mode."""
+        heat_area_mode = 0
+        if preset_mode == PRESET_DAY:
+            heat_area_mode = 1
+        elif preset_mode == PRESET_NIGHT:
+            heat_area_mode = 2
+
+        await self.coordinator.async_set_heat_area_mode(
+            self.heat_area_id, heat_area_mode
+        )
+
+        self.coordinator.data[self.heat_area_id]["HEATAREA_MODE"] = heat_area_mode
+        is_cooling = self.coordinator.get_cooling()
+        if heat_area_mode == 1:
+            if is_cooling:
+                self.coordinator.data[self.heat_area_id][
+                    "T_TARGET"
+                ] = self.coordinator.data[self.heat_area_id]["T_COOL_DAY"]
+            else:
+                self.coordinator.data[self.heat_area_id][
+                    "T_TARGET"
+                ] = self.coordinator.data[self.heat_area_id]["T_HEAT_DAY"]
+        elif heat_area_mode == 2:
+            if is_cooling:
+                self.coordinator.data[self.heat_area_id][
+                    "T_TARGET"
+                ] = self.coordinator.data[self.heat_area_id]["T_COOL_NIGHT"]
+            else:
+                self.coordinator.data[self.heat_area_id][
+                    "T_TARGET"
+                ] = self.coordinator.data[self.heat_area_id]["T_HEAT_NIGHT"]
+
+    @property
+    def extra_state_attributes(self):
+        """Return the extra state attributes.
+
+        is_locked (bool): Indicates whether the operating lock of the room operating panel is active.
+        lock_available (bool): Specifies whether a code is requested when the room operating unit is unlocked.
```

Comment:
`lock_available` doesn't explain what this attribute mean. If the attribute means whether or not a lock code is required, we should name it thereafter, eg `lock_code_required`.

I think both these attributes can be moved to separate binary sensor entities since they have a meaning regardless of the state of the climate device.

Reply:
I will remove extra state attributes, because I think they are rarely needed anyway.